src/io/state-file/schema.ts

```diff
@@ -1,5 +1,38 @@
 import JSZip from 'jszip';
 import { z } from 'zod';
+import { Vector3 } from '@kitware/vtk.js/types';
+import {
+  // IPiecewiseFunctionProxyMode,
+  PiecewiseGaussian,
+  PiecewiseNode,
+} from '@kitware/vtk.js/Proxy/Core/PiecewiseFunctionProxy';
+import {
+  CameraConfig,
+  SliceConfig,
+  WindowLevelConfig,
+  VolumeColorConfig,
+} from '../../store/view-configs/types';
+import { LPSAxisDir } from '../../types/lps';
+import { LayoutDirection } from '../../types/layout';
+import {
+  ViewType,
+  ColorBy,
+  ColorTransferFunction,
+  OpacityFunction,
+  OpacityGaussians,
+  OpacityPoints,
+  OpacityNodes,
+  ColoringConfig,
+  CVRConfig,
+} from '../../types/views';
+
+// This is duplicated from shims.vtk.d.ts as we can't use the enum values at
+// runtime as they are stripped.
+export enum IPiecewiseFunctionProxyMode {
+  Gaussians = 0,
+  Points = 1,
+  Nodes = 2,
+}
```

Comment:
I think you want to use `vtkPiecewiseFunctionProxy.Modes` instead of this enum, e.g. `vtkPiecewiseFunctionProxy.Modes.Gaussians`.

Reply:
Ok, good to know

Reply:
Great!

Reply:
This don't work, vtkPiecewiseFunctionProxy still refers to an enum defined in a d.ts file which will be stripped.

Reply:
I've used `vtkPiecewiseFunctionProxy.Modes.[Name]` as a value in other places, and it's backed by an actual object that maps those names to values. In what way is it getting stripped?

Reply:
So the issue is as I understand it that even with `vtkPiecewiseFunctionProxy.Modes` its type is defined as an `enum` ( `IPiecewiseFunctionProxyMode` ) inside `shim.vtk.d,ts`. Now unlike other types such as `interface` an enum in TypeScript has complied output ( the JavaScript implementation ). `.d.ts` are not explicitly imported so TypeScript does not have a place to inject the generated code. So if you try to refer to them at runtime for example `console.log(vtkPiecewiseFunctionProxy.Mode.Gaussians)` you will get a runtime error like this:

```
bootstrap:27 Uncaught TypeError: Cannot read properties of undefined (reading 'Mode')
    at ./src/io/state-file/schema.ts (schema.ts:38:13)
    at __webpack_require__ (bootstrap:24:1)
    at fn (hot module replacement:62:1)
    at ./src/store/datasets-images.ts (datasets-files.ts:11:8)
    at __webpack_require__ (bootstrap:24:1)
    at fn (hot module replacement:62:1)
    at ./src/store/datasets-dicom.ts (zip.ts:4:8)
    at __webpack_require__ (bootstrap:24:1)
    at fn (hot module replacement:62:1)
    at ./src/store/datasets.ts (datasets-models.ts:10:8)
```
If its back by a real object, it appears that TypeScript is still just using the type its finds in the `.d.ts` file.

Reply:
Oddly enough, I can do the following just fine:

```js
import vtkPiecewiseFunctionProxy from '@kitware/vtk.js/Proxy/Core/PiecewiseFunctionProxy';
console.log(vtkPiecewiseFunctionProxy.Mode.Gaussians);
```

I don't think I have any local changes that can change this behavior, but I can check out a new copy to see.

Reply:
Ok, I see the problem, I was adding to the import inside `{ } from '@kitware/vtk.js/Proxy/Core/PiecewiseFunctionProxy'`, using the default import syntax does indeed resolve the issue.